File: CrackingTheCodeInterview/Array&Strings/Python3/ZeroMatrix.py
def zeroMatrix(matrix):
	for i in range(len(matrix[0])):
		for j in range(len(matrix[0])):
			if matrix[i][j] == 0:
				#Note it will just work for n*n matrix
				for k in range(len(matrix[0])):
					matrix[i][k] = '-'
					matrix[k][j] = '-'
	for i in range(len(matrix[0])):
		for j in range(len(matrix[0])):
			if matrix[i][j] == '-': matrix[i][j] = 0
	return matrix
def print_matrix(matrix):
	print('----------------')
	for i in matrix: print(*i, sep='\t')

print_matrix(zeroMatrix([
[1, 2, 3, 0, 5],
[16, 17, 18, 19, 6],
[15, 24, 25, 20, 7],
[14, 0, 22, 0, 8],
[13, 12, 11, 10, 9],
]))

File: CrackingTheCodeInterview/Array&Strings/C++/ZeroMatrix.cpp
#include <bits/stdc++.h>
#if !defined(True)
#define True  (1==1)
#endif
#if !defined(False)
#define False (!True)
#endif
using namespace std;

void print_matrix(vector<vector<uint32_t> > matrix){
	cout << "----------------" << endl;
	for(auto line: matrix) {
		for (auto num: line) cout << num << '\t' ;
		std::cout << std::endl;
	}
}

vector<vector<uint32_t> > zeroMatrix(vector<vector<uint32_t> > matrix){
	for (size_t i = 0; i < matrix.size(); i++){
		for (size_t j = 0; j < matrix[0].size(); j++){
			if (matrix[i][j] == 0){
				for (size_t k = 0; k < matrix.size(); k++) matrix[k][j] = NULL;
				for (size_t k = 0; k < matrix[0].size(); k++) matrix[i][k] = NULL;
			}
		}
	}
	for (size_t i = 0; i < matrix.size(); i++)
		for (size_t j = 0; j < matrix[0].size(); j++)
			if(matrix[i][j] == NULL) matrix[i][j] = 0;
	return matrix;
}

int main(int argc, char const **argv){
	print_matrix(zeroMatrix({
		{1, 0, 3, 4, 5},
		{16, 17, 18, 0, 6},
		{15, 24, 25, 20, 7},
		{14, 0, 22, 21, 8},
		{0, 12, 11, 10, 9},
	}));
	return 0;
}